### PR TITLE
Ignore content length checks when the stream is encoded

### LINKED ISSUE #1055: Internal downloader throws `ContentLengthError`
```
$ ghcup --downloader=internal tui
[ Info  ] downloading: https://raw.githubusercontent.com/haskell/ghcup-metadata/master/ghcup-0.0.8.yaml as file /home/hasufell/.ghcup/cache/ghcup-0.0.8.yaml
[============================================================================================================================================] 141%
ghcup: ContentLengthError Nothing (Just 98256) 69674
```

### PR DESCRIPTION
**Pull request by dfordivam**
For `tar.xz` files the progress bar + length checks works as expected as they are sent by the server without any encoding.

The yaml files are usually gzipped encoded and the progress bar is not shown.

fixes #1055

### CODE REVIEW

File: lib/GHCup/Download/IOStreams.hs [hunk 1]
```diff
@@ -123,11 +123,12 @@ downloadInternal = go (5 :: Int)
 
     downloadStream r i' = do
       void setup
-      let size = case getHeader r "Content-Length" of
-            Just x' -> case decimal $ decUTF8Safe x' of
+      -- We can only do content length checks & progress bar when the stream is not encoded
```
Comment (hasufell) [inline lib/GHCup/Download/IOStreams.hs:126]: Afaiu  there's also `identity` encoding, which would work.

Comment (hasufell) [reply lib/GHCup/Download/IOStreams.hs:126]: See https://www.iana.org/assignments/http-parameters/http-parameters.xhtml

Comment (dfordivam) [reply lib/GHCup/Download/IOStreams.hs:126]: fixed

**Comment by dfordivam** [comment]
this is ready for review

**Pull request merged**